nats/aio/client.py

```diff
@@ -1437,38 +1341,38 @@ async def _process_msg(self, sid, subject, reply, data):
         self.stats['in_bytes'] += payload_size
 
         sub = self._subs.get(sid)
-        if sub is None:
+        if not sub:
             # Skip in case no subscription present.
             return
 
-        sub.received += 1
-        if sub.max_msgs > 0 and sub.received >= sub.max_msgs:
+        sub._received += 1
+        if sub._max_msgs > 0 and sub._received >= sub._max_msgs:
```

Comment:
I might be wrong but maybe should call `sub._stop_processing` here to mark the `_wait_for_msgs_task` for the subscriptions as done?